test/test_linalg.py

```diff
@@ -4920,18 +4922,24 @@ def test_addbmm(self, device, dtype):
         num_batches = 2
         M, N, O = 2, 3, 4
 
-        if self.device_type == 'cpu':
-            is_supported = True
-            if dtype == torch.bfloat16:
+        is_supported = True
+        is_cuda_bfloat = False
+        if dtype == torch.bfloat16:
+            if self.device_type == 'cpu':
                 self.precision = 1  # 43 vs 43.75
-        else:
-            is_supported = (dtype != torch.bfloat16 or AMPERE_OR_ROCM)
+            else:
+                is_cuda_bfloat = True
+                is_supported = TEST_WITH_ROCM or (CUDA11 and SM53)
 
         if not is_supported:
-            b1 = make_tensor((num_batches, M, N), device, dtype, low=-1, high=1)
-            b2 = make_tensor((num_batches, N, O), device, dtype, low=-1, high=1)
-            t = make_tensor((M, O), device, dtype, low=-1, high=1)
-            self.assertRaisesRegex(RuntimeError, "type|Type|not implemented|Ampere", lambda: torch.addbmm(t, b1, b2))
+            if not is_cuda_bfloat:
```

Comment:
See above suggestion on collapsing this for readability by testing the exact condition on one line.

Reply:
fixed